src/AVFoundation/AVContentKeySession.cs

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright 2021 Microsoft Corp
+//
+// Authors:
+//   Alex Soto (alexsoto@microsoft.com)
+//   Rachel Kang (rachelkang@microsoft.com)
+//
+
+#if false
+using System;
+using System.Runtime.InteropServices;
+using CoreGraphics;
+using ObjCRuntime;
+
+using CMSampleBufferRef = System.IntPtr;
+using AVContentKey = System.IntPtr;
+using NSErrorPtr = System.IntPtr;
+
+namespace AVFoundation {
+    public partial class AVContentKeySession {
+
+        [iOS (14, 5), Mac (11, 3), TV (14, 5), Watch (7,4)]
+        [DllImport (Constants.AVFoundationLibrary)]
+        [return: MarshalAs (UnmanagedType.I1)]
+        static extern /* BOOL */ bool AVSampleBufferAttachContentKey (
+            /* CMSampleBufferRef */ CMSampleBufferRef sbuf,
+            /* AVContentKey */ AVContentKey contentKey,
+            /* NSError * _Nullable * _Nullable */ out NSErrorPtr outError);
+        
+        [iOS (14, 5), Mac (11, 3), TV (14, 5), Watch (7,4)]
+        public static bool AttachContentKey (CMSampleBuffer sampleBuffer, AVContentKey contentKey, out NSError error)
```

Comment:
`using AVContentKey = System.IntPtr;`

yet you define `AVContentKey` as a binding later ? that seems incorrect

Reply:
hmmm... @dalexsoto @mandel-macaque should I rename the pointer to something like `AVContentKeyRef`? Or is there a greater issue here?

Reply:
@spouliot is correct, trick, remove the #if false and compile.

Reply:
Q: why did you pick up `AVContentKeySession` as the type ?

The function prefix is `AVSampleBuffer`.

It also applies on a `CMSampleBuffer` so an extension method seems better, e.g. it would be used like

```csharp
var x = new CMSampleBuffer ();
x.Attach (new AVContentKey(), out var _);
```

instead of 
```csharp
var x = new CMSampleBuffer ();
AVContentKeySession.AttachContentKey (x, new AVContentKey(), out var _);
```

Reply:
> Q: why did you pick up AVContentKeySession as the type ?

@spouliot good catch that was me, it was because the function lives in `AVContentKeySession.h` IIRC, I think your proposal makes more sense than binding it like this 👍